### PR TITLE
feat(web): force delete with shift key

### PR DESCRIPTION
**Pull request by martabal**
## Changes made in this PR

With this PR, you can delete assets without trashing them with Shift + Delete

### CODE REVIEW

File: web/src/lib/utils/actions.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { notificationController, NotificationType } from '$lib/components/shared-components/notification/notification';
+import { api } from '@api';
+import { handleError } from './handle-error';
+
+export type OnAssetDelete = (assetId: string) => void;
```
Comment (alextran1502) [inline web/src/lib/utils/actions.ts:5]: maybe change the name to `OnDelete` to keep the consistency with other types

**Comment by jrasm91** [comment]
On my operation system, you have `delete` and `shift delete`. Shift delete only bypasses the trash, not the confirmation modal as well. I don't think it is a good idea to bypass the modal by default. If anything we should add an option to not show the confirm modal again after the first time.

**Comment by martabal** [comment]
> On my ubuntu system, you have `delete` and `shift delete`. Shift delete only bypasses the trash, not the confirmation modal as well. I don't think it is a good idea to bypass the modal by default. If anything we should add an option to not show the confirm modal again after the first time.

image

What do you think about that ?
Should we add a setting in the user settings to enable / disable this modal ?

File: web/src/lib/components/photos-page/delete-asset-dialog.svelte [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    escape: void;
+  }>();
+</script>
+
+<ConfirmDialogue
+  title="Permanently Delete Asset{size > 1 ? 's' : ''}"
+  confirmText="Delete"
+  on:confirm={handleConfirm}
+  on:cancel={() => dispatch('cancel')}
+  on:escape={() => dispatch('escape')}
```
Comment (jrasm91) [inline web/src/lib/components/photos-page/delete-asset-dialog.svelte:34]: I don't think we need separate cancel vs escape events, just emit "close" or "cancel" both both of them.

File: web/src/lib/stores/preferences.store.ts [hunk 3]
```diff
@@ -96,3 +96,5 @@ export const albumViewSettings = persisted<AlbumViewSettings>('album-view-settin
   sortDesc: true,
   view: AlbumViewMode.Cover,
 });
+
+export const showDeleteModal = persisted<boolean>('delete-confirm-dialog', true, {});
```
Comment (jrasm91) [inline web/src/lib/stores/preferences.store.ts:100]: This seems fine for me. It can be _per client_.

**Review by jrasm91 — APPROVED** [review]

File: web/src/lib/stores/preferences.store.ts (hunk 3 above)
Comment (martabal) [reply web/src/lib/stores/preferences.store.ts:100]: Should we add a toggle in the user settings ?

Comment (jrasm91) [reply web/src/lib/stores/preferences.store.ts:100]: Sureeeeeee

**Pull request merged**